src/com/xilinx/rapidwright/design/DRC.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2021 Xilinx, Inc.
+ * All rights reserved.
+ *
+ * Author: Eddie Hung, Xilinx Research Labs.
+ *
+ * This file is part of RapidWright.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.xilinx.rapidwright.design;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.xilinx.rapidwright.design.drc.NetRoutesThruLutAtMostOnce;
+
+/**
+ * Parent DRC that executes a list of child DRCs, returning the sum of all failed checks.
+ */
+public class DRC {
+    interface DrcStyleFunction {
```

Comment:
Perhaps this interface should be declared in its own file, `design/drc/DrcStyleFunction.java` and explicitly declare its scope?

Reply:
This follows the same pattern as https://github.com/Xilinx/RapidWright/blob/3a6e8fc965926b1dd9bf30b936a052f32b46e9d6/src/com/xilinx/rapidwright/MainEntrypoint.java#L75-L77 in that it's a "functional interface" (rather than a "class interface") which allows it to represent static methods.

My earlier approach used an abstract class interface that forced subclasses to implement the `run` function, but this approach required (a) a new instance of the subclass to be created in order to run the DRC, and (b) reflection in order to get a handle to its `run` method to be invoked.

I believe this approach alleviates both of these problems.